lib/validator.js

```diff
@@ -0,0 +1,29 @@
+/*!
+ * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const validation = require('bedrock-validation');
+
+exports.validator = ({bodySchema, querySchema}) => {
+  if(!(bodySchema || querySchema)) {
+    throw new TypeError(
+      'One of the following parameters is required: ' +
+        '"bodySchema", "querySchema".');
+  }
+  return (req, res, next) => {
+    if(bodySchema) {
+      const result = validation.validateInstance(req.body, bodySchema);
+      if(!result.valid) {
+        return next(result.error);
+      }
+    }
+    if(querySchema) {
+      const result = validation.validateInstance(req.query, querySchema);
+      if(!result.valid) {
+        return next(result.error);
+      }
+    }
```

Comment:
Just a note here: When writing code like the above, it is good to consider which of these schema checks (query or body) is more complex/will be more costly in the common case and ensure we reject things quickly. I just wanted the author to note that the order may matter here in light of those kinds of considerations. We don't need to do anything now -- and maybe we don't have the insight/data to know which is better, I just want it to be a consideration on the author's mind if it wasn't already.